crates/common/tedge_config/src/system_services/managers/general_manager.rs

```diff
@@ -14,7 +14,8 @@ pub struct GeneralServiceManager {
 
 impl GeneralServiceManager {
     pub fn try_new(config_root: PathBuf) -> Result<Self, SystemServiceError> {
-        let init_config = SystemConfig::try_new(config_root.clone())?.init;
+        let init_config = SystemConfig::try_new(config_root.clone())?.init.unwrap(); // TODO throw
+                                                                                     // error
```

Comment:
I would simply revert the change making optional the field `SystemConfig::init`.